honeycomb/src/main/java/org/commonjava/o11yphant/honeycomb/HoneycombManager.java

```diff
@@ -228,6 +231,61 @@ public Span getActiveSpan()
         return null;
     }
 
+    public Timer getSpanTimer( String name )
+    {
+        SpanContext spanContext = (SpanContext) honeycombContextualizer.extractCurrentContext();
+        if ( spanContext != null )
+        {
+            Timer timer = spanContext.getTimer( name );
+            if ( timer == null )
+            {
+                timer = new Timer();
+                spanContext.putTimer( name, timer );
+            }
+            return timer;
+        }
+        return null;
```

Comment:
One suggestion here: as this method may return null, so maybe we should consider to use java.util.Optional to wrap it, and make all caller easily to check the obj it wrapped, especially for stream programming. This is a best practice from java 8 that let us avoid to return null anymore, which will always bring potential NPE.
So here maybe 
....
   return Optional.of(timer)
}
return Optional.empty()

And from caller:
Optional<Timer> timer = ...
if(timer.isPresent){
timer.get()....
}else{
...
}
So there will be no ugly null checking anymore in the future.

Reply:
Sounds good. I updated all 3 places.